### PR TITLE
Replace moco-agent API from HTTP to gRPC

### PR DESCRIPTION
**Pull request by kfyharukz**
## ⚠️ Warning

This pull request depends on the following pull request changes:
https://github.com/cybozu-go/moco-agent/pull/24

Merge the dependent pull requests, release moco-agent, and rerun the E2E tests.
Until then, the test of this pull request will continue to fail.

## Overview

Replace moco-agent API from HTTP to gRPC.

## Tests

E2E tests will not work properly without a Container Image with modified moco-agent code.

We can confirm the e2e tests working on the local machine:

```console
$ pwd # moco-agent repo
/path/to/dir/github.com/cybozu-go/moco-agent
$ git checkout main

$ docker build . -t moco-agent:dev # Create a container image with the modified code
$ kind load docker-image moco-agent:dev --name moco-e2e # Load the kind cluster
```

```console
$ pwd # moco repo
/path/to/dir/github.com/cybozu-go/moco

$ git diff # add the binary-copy-container-image options for controller
diff --git a/e2e/manifests/moco_controller.yaml b/e2e/manifests/moco_controller.yaml
index 5ed7b0e..e1a6161 100644
--- a/e2e/manifests/moco_controller.yaml
+++ b/e2e/manifests/moco_controller.yaml
@@ -9,3 +9,6 @@ spec:
       containers:
       - name: manager
         imagePullPolicy: Never
+        args:
+          - "--binary-copy-container-image"
+          - "moco-agent:dev"
$ cd e2e/
$ make shutdown-kind clean setup launch-kind
$ kind load docker-image moco-agent:dev --name moco-e2e
$ make test
...
•
Ran 3 of 3 Specs in 37.778 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 0 Skipped

$ kubectl get po -n e2e-test # working fine MySQL cluster
NAME                  READY   STATUS    RESTARTS   AGE
moco-mysqlcluster-0   4/4     Running   0          97s
moco-mysqlcluster-1   4/4     Running   0          97s
moco-mysqlcluster-2   4/4     Running   0          97s
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -86,6 +87,10 @@ manifests: $(CONTROLLER_GEN)
 generate: $(CONTROLLER_GEN)
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
+.PHONY: update-agentrpc
+update-agentrpc:
+	# TODO
```
Comment (masa213f) [inline Makefile:92]: Could you update this rule?

**Review by masa213f — CHANGES_REQUESTED** [review]
Please close the connections to agents here
https://github.com/cybozu-go/moco/blob/v0.7.0/controllers/mysqlcluster_controller.go#L181

and update this document.
https://github.com/cybozu-go/moco/blob/main/docs/example_mysql_cluster.md

File: Makefile [hunk 2]
```diff
@@ -86,6 +89,12 @@ manifests: $(CONTROLLER_GEN)
 generate: $(CONTROLLER_GEN)
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
+.PHONY: agentrpc
+agentrpc: $(PROTOC)
+	curl -sfL https://github.com/cybozu-go/moco-agent/releases/download/v$(MOCO_AGENT_VERSION)/agentrpc.proto -O
+	mv agentrpc.proto agentrpc/
+	bin/protoc --go_out=. --go_opt=paths=source_relative --go-grpc_out=. --go-grpc_opt=paths=source_relative agentrpc/agentrpc.proto
```
Comment (masa213f) [inline Makefile:96]: Let's use the variable.

```suggestion
	$(PROTOC) --go_out=. --go_opt=paths=source_relative --go-grpc_out=. --go-grpc_opt=paths=source_relative agentrpc/agentrpc.proto
```

Comment (kfyharukz) [reply Makefile]: 🙇

**Review by masa213f — APPROVED** [review]
LGTM

**Pull request merged**